python/hyperon/base.py

```diff
@@ -340,6 +380,19 @@ def parse(self, tokenizer):
         catom = self.cparser.parse(tokenizer.ctokenizer)
         return Atom._from_catom(catom) if catom is not None else None
 
+    def parse_err(self):
```

Comment:
All Python exceptions are wrapped into `Error`-expression in MeTTa. However, this is a way for passing unpredicted exceptions to MeTTa. If some error is typical and expected (as syntax mistakes), then it might be better to have a dedicated error message in MeTTa (maybe, without inclusion of references to the implementation language). OTOH, if there is a chance that `parse` will be called outside the MeTTa runner, then throwing a Python exception can be convenient.  Thus, I don't have a strong preference. And I didn't check this PR in practice, so I'm ok with having it as is and refining this later if necessary.

Reply:
I believe wrapping the parsing error into a `Error`-expression should be done inside `import` atom implementation (or other possible MeTTa expressions which interpret the MeTTa program as a text). This is a question about Python API and I believe we can choose either way. I think it is a minor thing, just think that exceptions are more natural to Python as a way of error processing.

Reply:
Yes, this is what I also meant. I agree that Python functions should throw exceptions if they are not direct implementations of grounded functions for MeTTa, and these exceptions can be caught and turned into specific `Error`-expressions in the grounded functions like `import!`. The only thing in this case is that if `parse` throws an exception for the main script, what will happen?

Reply:
Caller should decide whether to catch exception and handle error or allow exception be re-thrown further. In case of main script it should catch the exception and print the error to the user I believe.

Reply:
This is implemented in a3808f1